web3.js/src/publickey.ts

```diff
@@ -94,7 +120,13 @@ export class PublicKey extends Struct {
    * Checks if two publicKeys are equal
    */
   equals(publicKey: PublicKey): boolean {
-    return this._bn.eq(publicKey._bn);
+    if (typeof publicKey._bn === 'bigint') {
+      return this._bn === publicKey._bn;
+    } else {
+      // If it's not a bigint, the other type comes from an old library, so we
+      // just check the underlying buffer representation.
+      return this.toBuffer().equals(publicKey.toBuffer());
+    }
```

Comment:
This fixes the issue with the tests! Since we were getting an old `PublicKey` from `spl-token`, this needs to be more permissive with the checks. `Uint8Array` doesn't have an equality check, so this seemed like the safest way to check equality between `BN` and `bigint`, but could also future proof us in case we change the underlying representation again